lib/ae_mdw/db/mutations/name_transfer_mutation.ex

```diff
@@ -18,8 +19,11 @@ defmodule AeMdw.Db.NameTransferMutation do
             block_index: Blocks.block_index()
           }
 
-  @spec new(Names.name_hash(), Db.pubkey(), Txs.txi(), Blocks.block_index()) :: t()
-  def new(name_hash, new_owner, txi, block_index) do
+  @spec new(Node.tx(), Txs.txi(), Blocks.block_index()) :: t()
+  def new(tx, txi, block_index) do
+    name_hash = :aens_transfer_tx.name_hash(tx)
+    new_owner = :aens_transfer_tx.recipient_pubkey(tx)
+
```

Comment:
Like we discussed yday, we should probably get rid of any core calls inside mutations

Reply:
I understand your concern but this are only getters (no call to the chain state).

Reply:
I thought we had already agreed on this. The point of not having calls to core functions was that you wouldn't need to mock things around in tests to execute transaction. For this to work you would need to create a `transfer_tx` which contains a lot of fields that the mutation doesn't need.

Reply:
For this, perhaps we can add a new builder (e.g. `AeMdw.Db.Name.transfer_mutation(tx, txi, block_index)`) like we have been doing already

Reply:
No problem, I can clarify. I have agreed on not depending on the state of a chain, i.e on calls to Node modules that leads to a read from a cache or from the database. Like the example of not depending for now on dev mode. 

For any tx/record on the other hand, it is something that very simple to create (the `AeMdw.Db.name_claim_mutations` is different because it's a combination of mutations and requires calls to the chain state).

If it's something that could help creating the tests, I would gladly implement the factories for the tests and prevent adding more indirection to the code. I can get this task.

Reply:
I think simply saying "never call core on mutations" simplifies things, and avoids having these kinds of discussions. Whereas "no calls to core on the mutations except when that call doesn't require the chain state" or "except when it's something very simple to create" are more complex statements. But I know I'll never get any of my comments agreed on, and since productivity is the priority, I'm approving!

Reply:
You've given before a good contribution suggesting to move chain calls code to `name_claim_mutations` during our call.
Only for this point of using a record for the other mutations, I really think it's not needed to avoid indirection. I will gladly provide helpers for the tests. Thank you.